src/vivarium/framework/results/stratification.py

```diff
@@ -18,80 +19,47 @@
     exhaustive categories into which simulants can be assigned.
-
     This class includes a :meth:`stratify <stratify>` method that produces an
     output column by calling the mapper on the source columns.
-
     """
 
-    def __init__(
-        self,
-        name: str,
-        sources: list[str],
-        categories: list[str],
-        excluded_categories: list[str],
-        mapper: VectorMapper | ScalarMapper | None = None,
-        is_vectorized: bool = False,
-    ):
-        """
-        Assign a default `mapper` if none was provided and check for non-empty
-        `categories` and `sources` otherwise.
-
-        Parameters
-        ----------
-        name
-            Name of the stratification.
-        sources
-            A list of the columns and values needed as input for the `mapper`.
-        categories
-            Exhaustive list of all possible stratification values.
-        excluded_categories
-            List of possible stratification values to exclude from results processing.
-            If None (the default), will use exclusions as defined in the configuration.
-        mapper
-            A callable that maps the columns and value pipelines specified by the
-            `requires_columns` and `requires_values` arguments to the stratification
-            categories. It can either map the entire population or an individual
-            simulant. A simulation will fail if the `mapper` ever produces an invalid
-            value.
-        is_vectorized
-            True if the `mapper` function will map the entire population, and False
-            if it will only map a single simulant.
-
-        Raises
-        ------
-        ValueError
-            If no mapper is provided and the number of sources is not 1.
-        ValueError
-            If the categories argument is empty.
-        ValueError
-            If the sources argument is empty.
-        """
-        self.name = name
-        self.sources = sources
-        self._user_provided_mapper = mapper
-        self.is_vectorized = is_vectorized
-        self.mapper = self._get_vector_mapper(mapper, is_vectorized)
-        if not self.sources:
-            raise ValueError("The sources argument must be non-empty.")
-
-        self.categories = categories
-        if not self.categories:
-            raise ValueError("The categories argument must be non-empty.")
-
-        self.excluded_categories = excluded_categories
+    name: str
+    """Name of the stratification."""
+    sources: list[str]
+    """A list of the columns and values needed as input for the `mapper`."""
+    categories: list[str]
+    """Exhaustive list of all possible stratification values."""
+    excluded_categories: list[str]
+    """List of possible stratification values to exclude from results processing.
+    If None (the default), will use exclusions as defined in the configuration."""
+    mapping_function: VectorMapper | ScalarMapper | None
```

Comment:
Changing this name is not backwards compatible? Have you checked that this won't break downstream repos and models?

Reply:
changed this back and changed the "new" mapper attribute to `vectorized_mapper`